draft-ietf-quic-transport.md

```diff
@@ -594,6 +596,27 @@ supports, as shown below.
 ~~~
 {: #version-negotiation-format title="Version Negotiation Packet"}
 
+A server that generates a version negotiation packet SHOULD include one of the
+reserved version numbers (those following the pattern 0x?a?a?a?a).  This ensures
+that clients are able to correctly handle an unsupported version.
+
+The design of version negotiation permits a server to avoid maintaining state
+for packets that it rejects in this fashion.  However, when the server generates
+a version negotiation packet, it cannot randomly generate a reserved version
+number.  This is because the server is required to include the same value in its
+transport parameters.  To avoid the selected version number changing during
+connection establishment, the reserved version can be generated as a function of
+values that will be available to the server when later generating its handshake
+packets.
+
+A pseudorandom function that takes client address information (IP and port) and
+the client selected version as input would ensure that there is sufficient
+variability in the values that a server uses.
+
+A client MAY send a packet using a reserved version number.  This can be used to
+solicit a list of supported versions from a server.
```

Comment:
Oh, I'm not arguing it wouldn't be good for the ecosystem.  I'm just arguing that no self-interested client will ever do so, because it's incurring a latency penalty to itself.

However, it's an interesting point that if you're making a first-time connection you don't need the connection to come up quickly (because it's an alternative) and if you're making a subsequent connection you'll have cached info from previous actual connections.  That might actually be a good reason to *remove* the hint and say that clients SHOULD start with a reserved version ID if they have no cached info.